kimchi/src/circuits/constraints.rs

```diff
@@ -343,35 +355,74 @@ impl<F: FftField + SquareRootField> LookupConstraintSystem<F> {
-                    panic!("Multiple lookup tables are currently not supported");
+                // Get the max width of all lookup tables
+                let max_table_width = lookup_tables
+                    .iter()
+                    .fold(0, |max_width, LookupTable { data, .. }| {
+                        std::cmp::max(max_width, data.len())
+                    });
+
+                let max_num_entries = d1_size - (ZK_ROWS as usize) - 1;
+
+                let mut lookup_table = vec![Vec::with_capacity(d1_size); max_table_width];
+                let mut table_ids: Vec<F> = Vec::with_capacity(d1_size);
+                let mut non_zero_table_id = false;
+                for table in lookup_tables.iter() {
+                    let table_len = table.data[0].len();
+
+                    // Update table IDs
+                    if table.id != 0 {
+                        non_zero_table_id = true;
+                    }
+                    let table_id: F = gate::i32_to_field(table.id);
+                    table_ids.extend((0..table_len).map(|_| table_id));
+
+                    // Update lookup_table values
+                    for (i, col) in table.data.iter().enumerate() {
+                        if col.len() != table_len {
+                            return Err(LookupError::InconsistentTableLength);
+                        }
+                        lookup_table[i].extend(col);
+                    }
+
+                    // Fill in any unused columns with 0
+                    for lookup_table in lookup_table.iter_mut().skip(table.data.len()) {
+                        lookup_table.extend((0..table_len).map(|_| F::zero()))
+                    }
+                }
+
+                // Note: we use `>=` here to leave space for the dummy value.
+                if lookup_table[0].len() >= max_num_entries {
+                    return Err(LookupError::LookupTableTooLong {
+                        length: lookup_table[0].len(),
+                        maximum_allowed: max_num_entries - 1,
+                    });
                 }
 
-                let lookup_table = lookup_tables.into_iter().next().unwrap();
+                // For computational efficiency, we choose the dummy lookup value to be all 0s in
+                // table 0.
+                let dummy_lookup_value: Vec<F> = vec![];
+                let dummy_lookup_table_id = 0;
 
-                // get the last entry in each column of each table
-                let dummy_lookup_value: Vec<F> = lookup_table
-                    .data
-                    .iter()
-                    .map(|col| col[col.len() - 1])
-                    .collect();
-                let dummy_lookup_table_id = lookup_table.id;
+                // Pad up to the end of the table with the dummy value.
+                lookup_table
+                    .iter_mut()
+                    .for_each(|col| col.extend((col.len()..max_num_entries).map(|_| F::zero())));
```

Comment:
par_iter_mut?

also, why use `F::zero()` here? This is creating an assumption that the dummy value for the `table_id:0` is always going to be `0` right? It'd be better not to rely on this assumption here :o

Reply:
It's hard-coded as zero above.

Reply:
I think it'd be a good idea to add a debug assert somewhere to make sure this assumption doesn't change